src/__tests__/tokenList.test.ts

```diff
@@ -1,19 +1,79 @@
 import { expect } from "chai";
+import { ethers } from "ethers";
 
 import { fetchTokenList } from "../hooks/token";
 import { networkInfo } from "../networks";
 
-describe("fetchTokenList()", () => {
+beforeEach(() => {
+  jest.spyOn(window, "fetch").mockImplementation(() => {
+    return Promise.resolve({
+      json: () =>
+        Promise.resolve({
+          tokens: [
+            {
+              name: "Gnosis",
+              address: "0x6810e776880c02933d47db1b9fc05908e5386b96",
+              symbol: "GNO",
+              decimals: 18,
+            },
+            {
+              name: "Dai Stablecoin",
+              address: "0x6b175474e89094c44da98b954eedeac495271d0f",
+              symbol: "DAI",
+              decimals: 18,
+            },
+            {
+              name: "Spam Token",
+              address: "",
+              symbol: "SPAM",
+              decimals: 0,
+            },
+          ],
+        }),
+      status: 200,
+    } as any);
+  });
+});
+
+describe("Mainnet tokenlist", () => {
+  it("Should parse its response correctly", async () => {
+    const resultingTokens = await fetchTokenList(1);
+    const gnoAddress = ethers.utils.getAddress("0x6810e776880c02933d47db1b9fc05908e5386b96");
+
+    expect(resultingTokens).to.have.lengthOf(2);
+    expect(resultingTokens.get(gnoAddress)?.symbol).to.equal("GNO");
+    expect(resultingTokens.get(gnoAddress)?.address.toLowerCase()).eq(gnoAddress.toLowerCase());
+    expect(resultingTokens.get(gnoAddress)?.decimals).to.equal(18);
+  });
+
+  it("Should not crash on errors", async () => {
+    jest.spyOn(window, "fetch").mockImplementation(() => {
+      return Promise.reject("Unexpected error.");
+    });
+
+    const resultingTokens = await fetchTokenList(1);
+    // eslint-disable-next-line @typescript-eslint/no-unused-expressions
+    expect(resultingTokens).to.be.empty;
```

Comment:
Could you avoid the lint ignore by putting the await statement directly into the expect?

Reply:
The problem here is that the to.be.empty is not a function so it looks like a normal expression which is not used.
But in fact this actually does something (checking if the object ist empty). There are other tests with the same issue.

I found a more elegant solution which disabled this rule for all test files by including this in the eslint config:

```
{
    "overrides": [
        {
            "files": ["*.test.ts", "*.test.tsx"],
            "rules": {
                "no-unused-expressions": "off"
            }
        }
    ]
}

```

Reply:
Seems legit... go for it!